Featureno/nri

boxartist opened the pull request:
画面表示を修正しました。
それぞれの下線部の表示を修正しました。
アイコン画像を挿入しました。
ツールを使いインデントの修正をしました。

Commit:
NRIの指摘箇所を修正しました

Commit:
NRIの指摘箇所を修正しました

Commit:
指摘箇所を修正しました

choimake commented on index.html:
```diff
@@ -0,0 +1,500 @@
+  <body>
+    <header id="pagehead">
+      <div class="logo"> 
+        <a href="https://www.nri.com/jp">
```
ここのリンクも「#」で大丈夫です。

choimake commented on index.html:
```diff
@@ -0,0 +1,500 @@
+      <a href="#" class="ham">メニュー
+      </a> 
+    </header>
+    <p class="topicpath">
+      <a href="#">企業情報
+      </a> &gt; NRIの強み : NRIのデジタルビジネス戦略
+    </p>
```
「パンぐずリスト」のキーワードで検索して、この部分の実装を全体的に見直してみてください。

choimake commented on index.html:
```diff
@@ -0,0 +1,500 @@
+      </section>
+      <section class="articleBodyText">
+        <h2> 
+          <img src="image/bg:blue.jpg" width="22" height="22" alt=""> NRIのデジタルビジネス戦略
```
「:」を名前で使うのは、あんまり好ましくないです。（マシンによっては、git操作とかでバグったりします。
別の記号にかえてください。

choimake commented on index.html:
```diff
@@ -0,0 +1,500 @@
+          <img src="image/story06_img_01.jpg" width="900" height="730" alt="NRIのデジタルビジネス戦略" class="aligncenter"> 
+        </figure>
+      </section>
+      <section class="articltextimage">
```
classやid名ですが、単語の切れ目がはっきりわかるような命名規則の方が読みやすいので、
キャメルケース、スネークケース、ケバブケース、どれかの規則をページ全体に適応するようにしてください。

choimake commented on index.html:
```diff
@@ -0,0 +1,500 @@
+      </section>
+      <div class="link">
+        <p class="textalignleft"> 
+          <a href="#">＜ビジネスモデルの特徴
```
「＜」みたいな記号は、htmlではなくcssで実装するようにしましょう。
htmlは文章構造を示す言語であって、装飾を示す内容が含まれているのはよくないです。

choimake reviewed (commented):
確認しました。
いくつか修正して欲しい点がありますので、対応お願いします。

全画面でみたときに、いくつかのパーツが左に寄っています。
あと、メニューの文字の余白もオリジナルと異なっています。
![スクリーンショット 2020-07-20 8 45 41](https://user-images.githubusercontent.com/52233450/87888188-31d8fb00-ca66-11ea-9e62-623cfa57c0f5.png)

![スクリーンショット 2020-07-20 8 47 50](https://user-images.githubusercontent.com/52233450/87888204-5d5be580-ca66-11ea-993b-84d2a33ee346.png)

footerの余白もオリジナルと異なっています。
![スクリーンショット 2020-07-20 8 47 21](https://user-images.githubusercontent.com/52233450/87888208-68af1100-ca66-11ea-818f-7cb195cd6e92.png)

デザインが乱れています。
![スクリーンショット 2020-07-20 8 46 15](https://user-images.githubusercontent.com/52233450/87888383-9cd70180-ca67-11ea-8bab-8bb9830f0b1e.png)

全体的に言えることですが、リンクにフォーカスを当てた時に、下線部がでる、色が変わるといった装飾がオリジナルにはあるので、こちらも実装してください。
（ただし、headerのメニューが出たり消えたりする部分に関しては必要ありません）
![スクリーンショット 2020-07-20 8 48 45](https://user-images.githubusercontent.com/52233450/87888281-e7a44980-ca66-11ea-82a1-0aa72ebe48a3.png)

Commit:
再度修正しました

boxartist commented:
ご確認よろしくおねがいします。

choimake commented on common/css/style.css:
```diff
@@ -417,18 +467,21 @@
     margin: 0 auto;
-    padding: 40px;
+    padding: 40px 0 112px;
+    color: #333333;
```
ファイルの末尾には改行を入れておきましょう。（理由はPOSIX）

choimake replied:
~抽出した位置が悪いですね。もっと下のファイル末尾の話です。~

boxartist replied:
ご指摘有り難うございます。
二点質問があるのですが、全画面表示でズレが出るとの事なのですが、自分のパソコンで検証画面でもチェックして直してみたのですが。どうやって正しい全画面表示を見れますか？

もう一点
「抽出した位置が悪いですね。もっと下のファイル末尾の話です。」
の意味がわかりません。

boxartist replied:
ご指摘有り難うございます。
二点質問があるのですが、全画面表示でズレが出るとの事なのですが、自分のパソコンで検証画面でもチェックして直してみたのですが。どうやって正しい全画面表示を見れますか？

もう一点
「抽出した位置が悪いですね。もっと下のファイル末尾の話です。」
の意味がわかりません。

choimake replied:
487行目で現在コードが終わっていると思いますが、そこに入れてください。
```
color: #333333
} // この下に改行を入れるという意味です。

```

choimake commented on common/css/style.css:
```diff
@@ -274,85 +312,91 @@
     color: #999999;
 }
+#PageBody .info .pageback p a:hover {
+   text-decoration: underline;
```
ここみたいに、ちょこちょこインデントのサイズがズレているところがあるので修正してください。

choimake commented on common/css/style.css:
```diff
@@ -274,85 +312,91 @@
-    margin: 0 auto;
-    padding: 60px;
+    margin: 100px 212.5px;
+    /* padding: 60px; */
```
この手の実装のコメントは不要な場合確実に削除するようにしましょう。
後々コードを読む人が（自分含む）、このコメントアウトが必要なのか不要なのか判断がつかなくなり、可読性が落ちるからです。

choimake reviewed (commented):
コメントしましたので確認お願いします。

- cssをチェッカーにかけたところ、いくつかエラーがあったので修正してください。

- 全画面でみたときに画面が左に寄っています。
![スクリーンショット 2020-07-23 12 16 49](https://user-images.githubusercontent.com/52233450/88249655-0c572600-cce0-11ea-900a-1541cadbf9b5.png)

- マウスホバー時の挙動が実装されていません。
![スクリーンショット 2020-07-23 12 17 24](https://user-images.githubusercontent.com/52233450/88249726-458f9600-cce0-11ea-9504-1cf4432a0e0d.png)

- マウスホバー時の挙動が実装されていません。
- pdfのアイコンが表示されていません。
![スクリーンショット 2020-07-23 12 18 10](https://user-images.githubusercontent.com/52233450/88249743-55a77580-cce0-11ea-8bcc-e759300f6367.png)

- 矢印のアイコンには下線がつかないです。（元サイトがそういう仕様です)
![スクリーンショット 2020-07-23 12 17 39](https://user-images.githubusercontent.com/52233450/88249790-78398e80-cce0-11ea-9c3b-f1e86c76c3c7.png)

choimake commented:
DevToolを使って、画面の横幅を2000pxとかに変更してページを見ると、UIが左に寄っているのがわかるかと思います。

Commit:
ご確認お願いします。

画面を真ん中に表示できるようにしました。
ヘッダーの挙動をつけました。
インデントの修正をしました。
CSSチェッカーにかけました。
改行を入れました。
必要な要素に :before もしくは : after をつけました。

kouyanh commented:
<img width="1680" alt="スクリーンショット 2020-07-25 9 06 31" src="https://user-images.githubusercontent.com/14869217/88444118-6f6ac900-ce56-11ea-957a-923b726aaa42.png">
↑見本のサイトと違いますので、修正をお願いします！


<img width="1168" alt="スクリーンショット 2020-07-25 9 06 52" src="https://user-images.githubusercontent.com/14869217/88444127-74c81380-ce56-11ea-8cd1-1b1a963ba38e.png">
↑幅を小さくした歳、右寄せになってしまいましたので、センタリングを中心に見直しをお願いします！

Commit:
表示を修正しました。

全体表示が中央にくるようにアレンジしました。

boxartist commented:
全体表示が中央にくるようにアレンジしました。

yuki0501 commented:
拡大などなにもせずに見たのですが、画像のように下部にスクロールバーが出ます。
スクロールバーを一番右端にスクロールすると、不要な余白が出てきます。
無理に画面幅を設定していませんか？

![image](https://user-images.githubusercontent.com/60833384/88476254-fbbdde80-cf71-11ea-89fe-7c7444d0ccaf.png)

boxartist commented:
検証画面を2000にして、とにかく画面を中央に寄せることだけを考えてあがいていました。確かにスクロールが出ますね。すいません気付きませんでした。

yuki0501 commented:
見本と比べて、「企業情報」のタイトルと経営方針のすきまが空きすぎています。
余白も見本と完全に一致させる必要はありませんが、見比べて修正してください。

![image](https://user-images.githubusercontent.com/60833384/88476327-7b4bad80-cf72-11ea-9e89-077af406d38d.png)

Commit:
横スクロールの表示を消しました

boxartist commented:
全体的な表示をの修正をしました。

choimake commented on common/css/style.css:
```diff
@@ -1,45 +1,45 @@
-    position: fixed;
     top: 0;
-    left: 0;
+    left: 0;  
```
`;`以降のスペース入力は不要なので消しましょう。
（何かしらのツール使って、自動で消しちゃうのが楽で良いです）

choimake commented on common/css/style.css:
```diff
@@ -274,161 +316,193 @@
 }
 /* ------SNSボタン------ */
-#pagefoot .footlink {
+#Pagefoot {
+    /* width: 1600px;
+    margin: 0 auto; */
+}
```
styleの指定がないパラメータに関しては削除しましょう。(#Pagefootごと消してください）
不要コードは可読性を落とします。

choimake reviewed (commented):
コメントしましたので確認お願いします。

- cssのインデントで何箇所が乱れているところがあるので、修正してください
（ツール使うと楽です）

- マウスホバー時の色が変わる挙動が実装できていません。
![スクリーンショット 2020-07-26 21 15 30](https://user-images.githubusercontent.com/52233450/88478695-64628680-cf85-11ea-9eea-0aac542feb11.png)

- footerの下線が出現する挙動が、一番左のメニュー以外実装できていません。
![スクリーンショット 2020-07-26 21 16 02](https://user-images.githubusercontent.com/52233450/88478697-662c4a00-cf85-11ea-87f0-2dc1642f95e8.png)

- マウスホバー時の色が変わる挙動が実装できていません。
![スクリーンショット 2020-07-26 21 11 38](https://user-images.githubusercontent.com/52233450/88478699-66c4e080-cf85-11ea-9d2e-6bc6c5bf07a4.png)

- 画像のローディングに失敗しているというエラーがあります。
![スクリーンショット 2020-07-26 21 14 16](https://user-images.githubusercontent.com/52233450/88478700-675d7700-cf85-11ea-885c-6f3e24ddc134.png)

- ここからの以下のメニューのリンクに、元サイトでは存在する矢印アイコンが存在しません。（画像読み込みに失敗しているのが原因かと予想されます。）
![スクリーンショット 2020-07-26 21 14 49](https://user-images.githubusercontent.com/52233450/88478701-67f60d80-cf85-11ea-88d5-d5d7b50fac99.png)

- ホバー時の色が変わる挙動が実装されていません。
- アイコンが表示されていません。(画像読み込み失敗によるもの？）
![スクリーンショット 2020-07-26 21 15 22](https://user-images.githubusercontent.com/52233450/88478703-688ea400-cf85-11ea-83d5-c2adf264e1e4.png)

- 画像の横幅にスキマがあいています。元サイトではないはずです。
![スクリーンショット 2020-07-26 21 11 51](https://user-images.githubusercontent.com/52233450/88478704-69273a80-cf85-11ea-9c21-f310073d3eba.png)

- アイコン部分には下線は不要です。（元サイトではそういう挙動になっているはずです）
![スクリーンショット 2020-07-26 21 15 11](https://user-images.githubusercontent.com/52233450/88478705-6a586780-cf85-11ea-9c39-6719536b4ef3.png)

Commit:
修正しました。

画面表示を修正しました。
それぞれの下線部の表示を修正しました。
アイコン画像を挿入しました。
ツールを使いインデントの修正をしました。

kouyanh commented:
> 拡大などなにもせずに見たのですが、画像のように下部にスクロールバーが出ます。
> スクロールバーを一番右端にスクロールすると、不要な余白が出てきます。
> 無理に画面幅を設定していませんか？
> 
> ![image](https://user-images.githubusercontent.com/60833384/88476254-fbbdde80-cf71-11ea-89fe-7c7444d0ccaf.png)

こちらの件ですが、ブラウザの画面サイズを微妙に小さくしたうえで確認すると、再度横スクロールが発生するようです。
1280*800のサイズでも発生します。

Commit:
指摘箇所を修正しました

Viewportを追加しました
mainタグを入れました
bodyタグを追加しました
htmlとcssのチェッカーにかけ修正しました
画像の大きさをCSSで変更しました
ハンバーガーメニューの表示を変更しました

boxartist commented:
Viewportを追加しました
mainタグを入れました
bodyタグを追加しました
htmlとcssのチェッカーにかけ修正しました
画像の大きさをCSSで変更しました
ハンバーガーメニューの表示を変更しました

choimake commented on index.html:
```diff
@@ -0,0 +1,290 @@
+      </a> 
+    </div>
+    <nav class="topicpath">
+      <ul>
+        <li>
+          <a href="#">NRI トップ</a></li>
+        <li><a href="#">企業情報</a></li>
+        <li>NRIの強み：NRIのデジタルビジネス戦略</li>
+      </ul>
```
インデントに関してですが、一部コードとして読みにくい部分があります。
インデントの役割のひとつとして、親子関係を読み取りやすくするというものがあります。

例えば、
```
<li> 
  <a href="#">採用情報</a> 
</li>
```
だったら、`li`の子要素に`a`タグがあるというタグの関係性が字下げ（インデント）によって明示的に示されています。

上記のコードでインデントが機能していないと感じる部分は例えばここです。
```
<li>
          <a href="#">NRI トップ</a></li>
```
`li`タグの子要素に`a`タグがあることはわかりますが、閉じタグの字下げの位置の関係で、親子関係が読み取りにくいです。
コードを良く読めば、`a`タグの横の`/li`が閉じタグとして機能していることはたしかにわかるのですが、それはコードを読んだ結果であり、この場合は字下げを読んでも閉じタグがどれなのかは読み取れません。

今は小さなコードなので、インデントが乱れていても読めなくはないですが、これが大きめのhtmlコードになると「コードを頑張って読む」程度ではなかなか読み取れないレベルになってきます。

その時に字下げがちゃんと記載されているコードだと、その字下げの関係からコードを読み取ること可能になり、コードを読む上でこれが非常に助けになるのです。

コードを書く上でインデントは基本中の基本であり、必ずできるようになって欲しいところなので、この機会にきちんと扱えるようになってください。

（コードの全体で似たような箇所があるので、合わせて確認、修正お願いします）

choimake commented on index.html:
```diff
@@ -0,0 +1,290 @@
+    </div>
+  </main>
+</body>
+</html>
```
ファイルの末尾には改行を入れましょう。

Githubでファイル末尾に改行がないと、赤いマークが出ます。

何故かと言いますと、
「テキストファイルは本来（POSIX的には）必ず最後は改行コードで終わっていなければならない」というルールがあるからです。

それが守られていないときにGithubでは赤いマークという形で警告がでます。

以下、参考URLです。
https://www.cqpub.co.jp/interface/sample/200902/if02_038.pdf

標準規格やIEEE、POSIXなどについて一度確認してみてください。

choimake commented on index.html:
```diff
@@ -0,0 +1,290 @@
+      <img src="image/building.jpg" alt="NRIの強み"> </section>
+      <nav class="articleDetailbody">
+      <div class="share"> 
+        <a href="#">
+          <img src="image/icon-facebook.svg"  alt="facebookボタン">
+        </a> <a href="#">
+          <img src="image/icon-twitter.svg" alt="ツイッターボタン">
+        </a> <a href="#">
+          <img src="image/icon-linkedin.svg" alt="linkedin">
```
`src="image/icon-facebook.svg"  alt="facebookボタン">`の`"`と`alt`のスペースは2つになってますが、それ以下のところはスペース1つになっています。
このような余白の乱れはコードの可読性を落とします。きちんと揃えるようにしましょう。
（ツールとか使って自動でやると便利です。ていうか手動だとやってられないです）

choimake reviewed (commented):
コメントしました。

機能面に関してはOKです。
あとはコードのインデントや余白、改行などの細かい点についていくつか確認して欲しい点があります。
対応お願いします。

Commit:
インデントの修正をしました。

choimake commented on index.html:
```diff
@@ -0,0 +1,467 @@
+        <p id="copyright"> <small>Copyright(c) Nomura Research Institute, Ltd. All rights reserved.</small> </p>
+      </footer>
+    </div>
+  </main>
```
mainタグを閉じる位置が完全におかしいですね。
（footer巻き込んでます）

boxartist replied:
修正しました

choimake commented on index.html:
```diff
@@ -0,0 +1,467 @@
+      <div class="mainVisual">
+        <p>NRIの強み</p>
+        <h2>NRIのデジタルビジネス戦略</h2>
+      </div> <img src="image/building.jpg" alt="NRIの強み"> </section>
```
ここのインデントも修正が必要です。

boxartist replied:
修正しました

choimake commented on index.html:
```diff
@@ -0,0 +1,467 @@
+    <nav class="articleDetailbody">
+      <div class="share"> 
+        <a href="#">
+          <img src="image/icon-facebook.svg"  alt="facebookボタン">
+        </a> 
```
srcとaltのスペースが他と比べて大きいです。

boxartist replied:
修正しまし

choimake commented on index.html:
```diff
@@ -0,0 +1,467 @@
+    </div>
+    <nav class="topicpath">
+      <ul>
+        <li> <a href="#">NRI トップ</a></li>
+        <li> <a href="#">企業情報</a> </li>
+        <li>NRIの強み：NRIのデジタルビジネス戦略</li>
```
スペースがあったり無かったりとスタイルが統一されていません

boxartist replied:
修正しました

choimake commented on index.html:
```diff
@@ -0,0 +1,467 @@
+  <meta name="description" content="野村総合研究所（NRI）の公式ホームページです。NRIからの提言や調査・レポート、商品サービス、ITソリューション事例、IR情報、採用情報、CSR情報などを掲載しています。"> </head>
+<body>
+  <header id="Pagehead">
+    <div class="logo"> <a href="#">
+        <img src="image/logo.png" alt="NRIのロゴ">
+      </a> </div>
```
インデントが乱れています。

boxartist replied:
修正しました

choimake reviewed (commented):
コメントしましたので確認お願いします

Commit:
指摘箇所を修正しました

若干表示がずれていて、CSSも少し修正しました

Commit:
チェッカーにかけ、修正しました。

Commit:
インデントの修正をしました

boxartist commented:
インデントの修正をしました。ご確認よろしくお願いします。

choimake commented on index.html:
```diff
@@ -0,0 +1,460 @@
+    <section class="mainVisualtext">
+      <div class="mainVisual">
+        <p>NRIの強み</p>
+        <h2>NRIのデジタルビジネス戦略</h2>
+      </div> <img src="image/building.jpg" alt="NRIの強み"> 
```
`img`の位置がインデント的に気持ち悪いです。
以下の位置関係に修正して欲しいです。
```
  <h2>NRIのデジタルビジネス戦略</h2>
</div> 
<img src="image/building.jpg" alt="NRIの強み">

boxartist replied:
修正しました

choimake reviewed (commented):
コメントしましたので確認お願いします。

Commit:
一箇所修正しました。

Commit:
自動整形ツールを使って修正しました

Commit:
修正しました

choimake reviewed (approved):
対応ありがとうございました。LGTMです。

今回のPRをみてcommitの大きさが気になったので、「gitのcommitの粒度（大きさ）は小さくしましょう」というお話をします。

今現在だと「1つのコミットで複数の修正をするというやり方」をされていますが、これはあまり良いコミットのやり方ではないかなと感じます。

理由は色々ありますが抜粋すると
- 前のバージョンのコミットに戻したい時に、コミットでの変更が大きいと手戻りが大きくなる
- コードの変更差分をみた時に、何を変更したかが読みにくい
です。

なので、例えば、
- Aの追加
- Bの修正
- Cの修正

という変更がしたいという場合だったら、A、B、Cでそれぞれでcommitするようなやり方を試してみてください。

コミットの大きさはどれくらいが正しいの？っていうのは実は明確な正解はなく、個人によって色々なパターンがあります。
が、基本的には大きすぎるコミットは割と悲劇を産むことが多いので、僕は「1つの修正に対して1つのコミット」をすることをオススメしています。

ちょっと難しいかもですが、以下にコミットの単位についての記事がありますのでお手隙で確認してみてください。
https://qiita.com/jnchito/items/40e0c7d32fde352607be
